### ISSUE TITLE
I think there is a wrong!

### ISSUE THREAD

**Issue by rikyaa** [cuiziteng/Aleth-NeRF#22]
# Training Stage with Concealing Field
    if mode != 'test':
        if i_level == 0:
            accum_prod_dark = torch.cat([torch.ones_like(alpha[..., :1] * coarse_dark[0]),
                torch.cumprod(1.0 - alpha[..., :-1] + eps, dim=-1) * coarse_dark[1:]], dim=-1,)
        else:
            accum_prod_dark = torch.cat([torch.ones_like(alpha[..., :1]) * fine_dark[0],
                torch.cumprod(1.0 - alpha[..., :-1] + eps, dim=-1) * fine_dark[1:]], dim=-1,)
here :**torch.cumprod(1.0 - alpha[..., :-1] + eps, dim=-1) * fine_dark[1:]**维度是（4096，64）（128，）没办法相乘的呀？期待你的回答

**Comment by rikyaa** [comment]
Author I'm very sorry to bother you, your article is very well written but the code you uploaded I double checked still think there is a dimensional mismatch here, looking forward to your answer, hopefully it will solve the confusion I've been having.

**Comment by cuiziteng** [comment]
您好 感谢关注 您可以print一下看看维度 fine那里的维度也是128 您可以自己跑一下代码 不会报错的

**Comment by rikyaa** [comment]
![batchsize](https://github.com/user-attachments/assets/7e00589c-09dc-48a0-b5e4-6d1e2ecddf8d)
![fine](https://github.com/user-attachments/assets/4d6fee3c-591d-4bdf-b5d1-373d1feff0e7)
![wrong](https://github.com/user-attachments/assets/5874e835-b9a0-49c2-98fa-1b492a9b0456)
![image](https://github.com/user-attachments/assets/9ae77876-2703-4a50-bcca-84a89199cb36)
![image](https://github.com/user-attachments/assets/7c09c3d2-94e0-4146-85b8-fadb9dbc43ad)
![image](https://github.com/user-attachments/assets/60662c93-1ee5-4f35-a24e-1bd60bff2f8d)
![image](https://github.com/user-attachments/assets/d33804d6-9d91-4b37-9e72-9a8e2d275010)
![image](https://github.com/user-attachments/assets/7e85ab6b-bff8-40dd-a0fa-a51911b92721)
![image](https://github.com/user-attachments/assets/8a9b3f01-04ba-4890-930b-637454aaad8a)

Hello author, your paper is very excellent. I want to make changes to the code. I modified fine_dark for the problem posted earlier, this time I re-downloaded the code and only modified the batchsize.I debugged the code you wrote again, but there is still a problem. I still think that the latitude in else does not match. torch.cumprod(1.0 - alpha[..., :-1] + eps, dim=-1)  latitude is (32, 64), and fine_dark[1:]  is (192,). This question has puzzled me for a long time, so I wanted to ask you. Did I understand the problem or something else? I look forward to your answer.
作者您好，你的论文很优秀我想对代码做出修改，前面发的问题我修改了fine_dark，这次我重新下载了代码只修改了batchsize我再次调试了你写的代码，还是存在问题，我还是觉得else这里面纬度是不匹配的，这个纬度是(32，64)，这里是(192，)，这个问题困惑了我很久希望所以想来问问你，是我理解出了问题还是其他的呢，期待您的解答

**Comment by cuiziteng** [comment]
您好，else这里表示的是通过NeRF的fineMLP, fineMLP最后的输出尺度不是64而是192，我也不清楚您这里是怎么出现的出来(32, 64)这样的情况，理论上讲理应是(32, 192)。为了验证是否是我的代码有问题，我这边重新clone了一下代码，并且重新跑了一遍，完全没有问题，问一下您这里是根据指令跑代码了嘛，还是只是在py文件调试出错呢？

<img width="949" alt="image" src="https://github.com/user-attachments/assets/cb5878db-1050-4eec-9718-a952e1164797">
<img width="375" alt="image" src="https://github.com/user-attachments/assets/09a1c3e9-5744-450a-a7bf-c48eb9e1ad14">
<img width="1037" alt="image" src="https://github.com/user-attachments/assets/f5959ee2-7b7f-446e-a57e-315c9d5f5422">

**Comment by rikyaa** [comment]
谢谢年的回复，调试的时候fineMLP最后的输出尺度确实是192，torch.cumprod(1.0 - alpha[..., :-1] + eps, dim=-1)是 (4096, 64), and fine_dark[1:] 是 (192,)我是调试的时候出现的这个纬度不匹配的问题，虽然运行代码CUDA_VISIBLE_DEVICES=0,1,2,3 python3 run.py --ginc configs/LOM/aleth_nerf/aleth_nerf_buu.gin --logbase ./logs --con 12 --eta 0.45不会报错而且正常运行，我再看看到底是怎么回事把

**Comment by rikyaa** [comment]
谢谢你的回复，问题已经解决了，代码没有问题，是我修改代码的过程中存在错误，感谢！！！
